File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    let buy = prices[0];
    let maxProfit = 0;

    for (let i = 1; i < prices.length; i++)
    {
        if (prices[i] < buy)
        {
            buy = prices[i];
        } else {
            let profit = prices[i] - buy;
            
            if (profit > maxProfit) {
                maxProfit = profit;
            }
        }
    }
    return maxProfit;
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int buy = prices[0];
        int maxProfit = 0;

        for (int i = 1; i < prices.length; i++)
        {
            if (prices[i] < buy)
            {
                buy = prices[i];
            } else 
            {
                int profit = prices[i] - buy;
                if (profit > maxProfit) {
                    maxProfit = profit;
                }
            }
        }
        return maxProfit;
    }
}